oak_functions/loader/src/server.rs

```diff
@@ -145,13 +144,25 @@ pub trait OakApiNativeExtension {
     /// Performs any cleanup or terminating behavior necessary before destroying the WasmState.
     fn terminate(&mut self) -> anyhow::Result<()>;
 }
+
 pub trait ExtensionFactory {
     fn create(&self) -> anyhow::Result<BoxedExtension>;
 }
 
-pub type BoxedExtension = Box<dyn OakApiNativeExtension + Send + Sync>;
+pub enum Extension {
+    Native(Box<dyn OakApiNativeExtension + Send + Sync>),
+    Uwabi(Box<dyn UwabiExtension + Send + Sync>),
+}
+
+pub type BoxedExtension = Extension;
 pub type BoxedExtensionFactory = Box<dyn ExtensionFactory + Send + Sync>;
 
+/// Trait for implementing an extension which relies on uwabi.
```

Comment:
Should it be

```suggestion
/// Trait for implementing an extension which relies on UWABI.
```

?